### DISCUSSION TITLE
Version 4.11.0-0.okd-2022-12-03-100655 not found in channel stable-4

### DISCUSSION THREAD

**Discussion by jeffmccune** [okd-project/okd#1435] [Q&A] [answered]
Over the weekend I got the update recommendations from `4.11.0-0.okd-2022-11-19-050030` to `4.11.0-0.okd-2022-12-02-145640` then again to `4.11.0-0.okd-2022-12-03-100655`

This morning, I got the following alert:

```
Version 4.11.0-0.okd-2022-12-03-100655 not found in channel stable-4. To request update recommendations, configure a channel that supports your version.
```

Any suggestions for how to fix the alert and get update recommendations working again?

**Comment by vrutkovs** [comment, marked as answer]
See https://github.com/okd-project/okd/issues/1433. 2022-12-03-100655 is a nightly, so you're recommended to revert back to [4.11.0-0.okd-2022-12-02-145640](https://amd64.origin.releases.ci.openshift.org/releasestream/4-stable/release/4.11.0-0.okd-2022-12-02-145640)

**Reply by jeffmccune** [reply]
Reverting back using the following command and the additional information at [Updating a cluster using the CLI](https://docs.okd.io/4.11/updating/updating-cluster-cli.html):

```sh
oc adm upgrade --to-image=quay.io/openshift/okd@sha256:924958ed5ba93ea9d72e4b5a319d0d4df220edbb891816c106215cbf318a0a40 --allow-explicit-upgrade
```

```txt
warning: The requested upgrade image is not one of the available updates.You have used --allow-explicit-upgrade for the update to proceed anyway
Requesting update to release image quay.io/openshift/okd@sha256:924958ed5ba93ea9d72e4b5a319d0d4df220edbb891816c106215cbf318a0a40
```

**Reply by vrutkovs** [reply]
yep, you'd probably need to add `--force` too

**Reply by jeffmccune** [reply]
Thanks, FYI the revert back to `4.11.0-0.okd-2022-12-02-145640` was successful without `--force`

**Reply by norbert-marx** [reply]
I run into the same issue. With the removal of 4.11.0-0.okd-2022-12-03-100655 I'm now blocked in any direction because one of the control nodes rebooted and doesn't start anymore because of missing docker images:

Trying to pull registry.ci.openshift.org/origin/4.11-2022-12-03-100655@sha256:7c7df602ae311bd172550bfba3b944ae157c145dd67e3ec02d081946c5d13077...
Error: initializing source docker://registry.ci.openshift.org/origin/4.11-2022-12-03-100655@sha256:7c7df602ae311bd172550bfba3b944ae157c145dd67e3ec02d081946c5d13077: reading manifest sha256:7c7df602ae311bd172550bfba3b944ae157c145dd67e3ec02d081946c5d13077 in registry.ci.openshift.org/origin/4.11-2022-12-03-100655: manifest unknown: manifest unknown

So I believe docker://registry.ci.openshift.org/origin/4.11-2022-12-03-100655@sha256:7c7df602ae311bd172550bfba3b944ae157c145dd67e3ec02d081946c5d13077 should be available that clusters are able to reboot nodes and reach a working state.

**Comment by ssams** [comment]
I've ran into this and our cluster accidentally got upgraded to `4.11.0-0.okd-2022-12-03-100655`. I'm trying to revert it back to `4.11.0-0.okd-2022-12-02-145640` as instructed above, and the procedure starts.

However it gets stuck at some point as some operators refuse to proceed, as the old rollout is not considered healthy: due to #1310 I had to restart some nodes as they were not responsive anymore (before initiating the downgrade), but now they fail to fully start as they try to pull some images for `4.11.0-0.okd-2022-12-03-100655` which seem to be not available anymore. So e.g. the *network* cluster operator does not proceed with the downgrade as it's waiting for all *multus* Pods to be running, still using the old non-available images. Several other operators downgraded successfully already, I hoped that I can get the temporary failure state of networking etc. too, but doesn't seem to be the case.

Any hints on how to get the cluster back into an operational state and convince the remaining operators to proceed with the downgrade?

**Reply by ssams** [reply]
I managed to get my cluster back to the stable release, in case anyone else is having similar issues here's what I did:
* initiated the downgrade with the `oc adm upgrade ...` command from the previous comments. Several operators were rolled back, but as indicated in my previous comment some were stuck due to restarted nodes which failed to pull the non-available images from the nightly release at this point.
* container runtime: first some of the missing images were preventing the start of kubelet at all on the restarted nodes -> checked all nodes whether the kubelet service was running (service "nodeip-configuration" as a dependency also blocked the start on some). updated any references to images pulling from `registry.ci.openshift.org/origin/4.11-2022-12-03-100655` to `quay.io/openshift/okd-content` in the following files on the nodes:
  * /etc/systemd/system/nodeip-configuration.service
  * /etc/crio/crio.conf.d/00-default
  * /etc/systemd/system/kubelet.service
  
  followed by a restart of the modified services via systemctl (a complete reboot should also work at this point). luckily, the `2022-12-03` (nightly) release seems to reuse many images from the `2022-12-02` (stable) release, so the image digests can be left untouched and work with the default release registry, so for example in the listed files I could simply replace images like `registry.ci.openshift.org/origin/4.11-2022-12-03-100655@sha256:7c7df602ae311bd172550bfba3b944ae157c145dd67e3ec02d081946c5d13077` with `quay.io/openshift/okd-content@sha256:7c7df602ae311bd172550bfba3b944ae157c145dd67e3ec02d081946c5d13077`
* at this point kubelet was running again on all nodes, but containers still didn't start due to missing cluster networking -> I manually edited the existing Pods in the `openshift-multus` and `openshift-sdn` namespaces (i.e. any that were failing to start due to image pull errors). similar to the previous file update, changing all image registries from `registry.ci.openshift.org/origin/4.11-2022-12-03-100655` to `quay.io/openshift/okd-content` but keeping the digests was sufficient (must be changed for all containers and init containers in the pods). Note that I edited the Pods directly in the hope that a future rollout/update of the DaemonSet which should actually manage them would revert my changes back to the desired correct configuration (which it did later during this process as I had hoped), whilst leaving the deployment on the still running nodes untouched.

  As I couldn't access the web console anymore whilst the networking was broken, I did these changes directly on one of the still running master nodes -> access via SSH and switch to root, `export KUBECONFIG="/etc/kubernetes/static-pod-resources/kube-apiserver-certs/secrets/node-kubeconfigs/localhost.kubeconfig"` to get access to the kube-apiserver (not sure if there's another simpler location of a working kubeconfig, but this one worked for me), then `kubectl` could be used to edit the pods as normal, changes were picked up properly by the remote kubelets without further actions already at this point.
* This was sufficient so the update could proceed - there were further image pull errors for example in the `openshift-dns` namespace, but these fixed themselves after a while as the upgrade progressed again.
* after all other operators were "upgraded" to the desired stable release again, the only one left in a broken state was the machine config operator, which complained about the on-disk changes I did on the nodes earlier. Hence I temporarily undid the manual changes and restored the original files from the above step (switching back to the broken `registry.ci.openshift.org/origin/4.11-2022-12-03-100655` in the kubelet etc. config files) -> as soon as the original files were restored, the machine config proceeded with the update and changed it to the desired stable registry again (including the usual node reboot, but all nodes started with the correct release registry again)

At this point the complete cluster looks good again, so far I've seen no traces of the broken registry anymore and all operators are marked as available with `4.11.0-0.okd-2022-12-02-145640` shown as cluster version.
